include/nudb/error.hpp

```diff
@@ -8,75 +8,12 @@
 #ifndef NUDB_ERROR_HPP
 #define NUDB_ERROR_HPP
 
-#include <boost/system/system_error.hpp>
 #include <boost/system/error_code.hpp>
 
 namespace nudb {
 
-/// The type of system-specific error code returned by the implementation
-#if GENERATING_DOCS
-class error_code{};
-
-#else
-using boost::system::error_code;
-
-#endif
-
-/// The type of cross-platform error code used by the implementation
-#if GENERATING_DOCS
-class error_condition{};
-
-#else
-using boost::system::error_condition;
-
-#endif
-
-/// The type of system-specific exception used when throwing
-#if GENERATING_DOCS
-class system_error{};
-
-#else
-using boost::system::system_error;
-
-#endif
-
-/// Returns the  category used for system-specific error codes
-#if GENERATING_DOCS
-error_category const&
-system_category();
-
-#else
-using boost::system::system_category;
-
-#endif
-
-/// Returns the category used for cross-platform error codes
-#if GENERATING_DOCS
-error_category const&
-generic_category();
-
-#else
-using boost::system::generic_category;
-
-#endif
-
-/// The base class used for error categories
-#if GENERATING_DOCS
-class error_category{};
-
-#else
-using boost::system::error_category;
-
-#endif
-
-/// The set of constants used for cross-platform error codes
-#if GENERATING_DOCS
-enum errc{};
-
-#else
-namespace errc = boost::system::errc;
-
-#endif
+// lift all error code stuff from boost::system into the nudb namespace
+using namespace boost::system; // system_category, errc
```

Comment:
Do we really want to import the entire namespace? Then any chances to boost system now become part of the public interface of NuDB. That was not the intention of the original type aliases. The intent was to give users a small number of well defined aliases which in the future could be switched to, say, their `std::` equivalents. Its not clear how this change improves the documentation...